botorch/models/approximate_gp.py

```diff
@@ -137,7 +137,13 @@ def train(self: TApproxModel, mode: bool = True) -> TApproxModel:
         return Module.train(self, mode=mode)
 
     def posterior(
-        self, X, output_indices=None, observation_noise=False, *args, **kwargs
+        self,
+        X,
+        output_indices=None,
+        observation_noise=False,
+        posterior_transform=None,
+        *args,
+        **kwargs,
```

Comment:
Can we remove these catch-alls as well? This was the reason this for this being a silent bug.
```suggestion
```